lib/avro_turf/messaging.rb

```diff
@@ -228,6 +234,14 @@ def fetch_schema_by_id(schema_id)
       [schema, schema_id]
     end
 
+    def fetch_schema_by_body(schema_name:, subject: nil, namespace: nil)
+      schema = @schema_store.find(schema_name, namespace)
+      schema_data = @registry.check(subject || schema.fullname, schema)
```

Comment:
If I am not wrong, this method `check` will actually perform a POST request to the registry.
So if you passed the credentials, it will register the schema if it doesn't exist.
So the 'read_only' param will actually not be true

Reply:
The check method does a different call to the API than the register one, and this call does not register the schema… it’s only a post because it needs to pass the schema in the body of the request

Reply:
Ah, totally right!